src/main/java/tds/content/repositories/impl/S3ItemDataRepository.java

```diff
@@ -37,6 +38,7 @@
 
 @Repository
 @Primary
```

Comment:
You are correct - @Primary is ignored with profiles specified. In fact I'll be switching the annotations around so that if no profile is defined, we will default to the S3 implementations.